src/CoreShop/Bundle/ResourceBundle/DependencyInjection/CoreShopResourceExtension.php

```diff
@@ -75,71 +78,41 @@ public function prepend(ContainerBuilder $container)
             ]
         ];
 
-        $frameworkConfig = [
-            'form' => true
-        ];
+        if (count($fosRestConfigs) > 1) {
+            $configMerger = new ConfigMerger();
 
-        $doctrineConfig = [
-            'dbal' => [
-                'mapping_types' => [
-                    'enum' => 'string'
-                ]
-            ],
-            'orm' => [
-                'auto_generate_proxy_classes' => '%kernel.debug%',
-                'entity_managers' => [
-                    'default' => [
-                        'auto_mapping' => true,
-                        'metadata_cache_driver' => [
-                            'type' => 'service',
-                            'id' => 'doctrine_cache.providers.coreshop_pimcore_metadata_cache'
-                        ],
-                        'result_cache_driver' => [
-                            'type' => 'service',
-                            'id' => 'doctrine_cache.providers.coreshop_pimcore_result_cache'
-                        ],
-                        'query_cache_driver' => [
-                            'type' => 'service',
-                            'id' => 'doctrine_cache.providers.coreshop_pimcore_query_cache'
-                        ]
-                    ]
-                ]
-            ]
-        ];
+            $restConfigs = [];
+            foreach ($fosRestConfigs as $rest) {
+                if (!is_array($rest)) {
+                    continue;
+                }
 
-        $doctrineCacheConfig = [
-            'custom_providers' => [
-                'coreshop_pimcore_cache' => [
-                    'prototype' => 'coreshop.doctrine.cache.pimcore'
-                ]
-            ],
-            'providers' => [
-                'coreshop_pimcore_metadata_cache' => [
-                    'coreshop_pimcore_cache' => []
-                ],
-                'coreshop_pimcore_result_cache' => [
-                    'coreshop_pimcore_cache' => []
-                ],
-                'coreshop_pimcore_query_cache' => [
-                    'coreshop_pimcore_cache' => []
-                ]
-            ]
-        ];
+                $restConfigs = $configMerger->merge($restConfigs, $rest);
+            }
 
-        $stofDoctrineExtensions = [
-            'default_locale' => '%locale%',
-            'orm' => [
-                'default' => [
-                    'timestampable' => true
-                ]
-            ]
-        ];
+            $fosRestConfigs = [$restConfigs];
+
+            $this->setExtensionConfig($container, 'fos_rest', $fosRestConfigs);
```

Comment:
have you tested that without any additional custom config? not sure if thats working.